packages/create-emotion-styled/types/test.tsx

```diff
@@ -131,9 +131,9 @@ const StyledTag1 = createStyled('label')`
 const StyledCompWithButton = StyledClassComp0.withComponent('button');
 const StyledCompWithFunComp2 = StyledFunComp0.withComponent(TestFunComp1);
 
-<StyledCompWithButton />;
```

Comment:
Should we keep this case?

Reply:
@tkh44 This is the problem what I mentioned in #721.
We cannot distinguish props that is used to make style and props that is passed to internal component without type parameter, so there's two ways, considering them as props for style, and considering them as a internal-only props.

#721, @mvestergaard said complaint about second choice (my original choice), so I changed this into first one.

Reply:
@tkh44 Could you discuss about this with @mvestergaard ?

Reply:
This might be naive, but wouldn't this make that use case work?
```typescript
withComponent<T extends keyof JSX.IntrinsicElements>(
     tag: T,
     options?: StyledOptions,
   ): StyledOtherComponent<JSX.IntrinsicElements[T], JSX.IntrinsicElements[T], Theme>;
```

Basically force Props to be the same as InnerProps.

Reply:
@mvestergaard The problem occurs in original style. Original style already depends on old `Props`, so we cannot change them freely. If we do so, it will break the type.

Reply:
Anyway, I'm not sure what i can discuss with @tkh44 
I've spoken my case on #721 already. The gist is basically, that a breaking change was made. Something that worked just fine before, now requires explicit typing.
In my head this is wrong, but I can't force you guys to change it based on that. So you need to decide internally how you intend the library to be used.
The most common use case should be prioritized.

My personal opinion is that restyling components:
```
const Restyled = styled(OtherComponent)``
```
is a more common use case than switching the internal component
```
const SomeStyledThing = styled.div``;
const SomeOtherStyledThing = SomeStyledThing.withComponent('span');
```
But @Ailrun believes it's the other way around

Reply:
@mvestergaard Thank you for your summary!
What I mean with 'discussion' is about 'Which one is more common and preferable use case'.
How about your opinion, @tkh44?